tribe-ext-limit-week-view-time-range.php

```diff
@@ -0,0 +1,266 @@
+
+		/**
+		 * Setup the Extension's properties.
+		 *
+		 * This always executes even if the required plugins are not present.
+		 */
+		public function construct() {
+			$this->add_required_plugin( 'Tribe__Events__Pro__Main' );
+
+			// Set the extension's TEC URL
+			$this->set_url( 'https://theeventscalendar.com/extensions/limit-the-time-range-in-week-view/' );
+		}
+
+		/**
+		 * Get this plugin's options prefix.
+		 *
+		 * Settings_Helper will append a trailing underscore before each option.
+		 *
+		 * @see \Tribe\Extensions\Example\Settings::set_options_prefix()
+		 *
+		 * @return string
+		 */
+		private function get_options_prefix() {
+			return (string) str_replace( '-', '_', PLUGIN_TEXT_DOMAIN );
+		}
+
+		/**
+		 * Get Settings instance.
+		 *
+		 * @return Settings
+		 */
+		private function get_settings() {
+			if ( empty( $this->settings ) ) {
+				$this->settings = new Settings( $this->get_options_prefix() );
+			}
+
+			return $this->settings;
+		}
+
+		/**
+		 * Extension initialization and hooks.
+		 */
+		public function init() {
+			load_plugin_textdomain( 'PLUGIN_TEXT_DOMAIN', false, basename( dirname( __FILE__ ) ) . '/languages/' );
+
+			if ( ! $this->php_version_check() ) {
+				return;
+			}
+
+			if ( ! $this->v2_views_check() ) {
+				return;
+			}
+
+			$this->class_loader();
+
+			$this->get_settings();
+
+			add_filter( 'tribe_events_week_get_hours', [ $this, 'filter_week_hours' ] );
+		}
+
+		/**
+		 * Check if we have a sufficient version of PHP. Admin notice if we don't and user should see it.
+		 *
+		 * @link https://theeventscalendar.com/knowledgebase/php-version-requirement-changes/ All extensions require PHP 5.6+.
+		 *
+		 * @return bool
+		 */
+		private function php_version_check() {
+			$php_required_version = '5.6';
+
+			if ( version_compare( PHP_VERSION, $php_required_version, '<' ) ) {
+				if (
+					is_admin()
+					&& current_user_can( 'activate_plugins' )
+				) {
+					$message = '<p>';
+					$message .= sprintf( __( '%s requires PHP version %s or newer to work. Please contact your website host and inquire about updating PHP.', PLUGIN_TEXT_DOMAIN ), $this->get_name(), $php_required_version );
+					$message .= sprintf( ' <a href="%1$s">%1$s</a>', 'https://wordpress.org/about/requirements/' );
+					$message .= '</p>';
+
+					tribe_notice( PLUGIN_TEXT_DOMAIN . '-php-version', $message, [ 'type' => 'error' ] );
+				}
+
+				return false;
+			}
+
+			return true;
+		}
+
+		/**
+		 * Check if we have the needed view. Admin notice if we don't and user should see it.
+		 *
+		 * @return bool
+		 *
+		 */
+		private function v2_views_check() {
+			$view_required_version = $this->view_needed;
+
+			$view_breaks_version = $view_required_version == 'V1 legacy' ? 'V2 updated' : 'V1 legacy';
+
+			$show_warning = (
+				$view_required_version == 'V1 legacy' && tribe_events_views_v2_is_enabled()
+				|| $view_required_version == 'V2 updated' && ! tribe_events_views_v2_is_enabled()
```

Comment:
`$view_required_version` comes from `$this->view_needed`, which is set to `v1`, why are we checking if its value is v2? I don't see us changing this value, maybe I'm missing something?